### PR TITLE
Fixed #32627 -- Fixed QuerySet.values()/values_list() crash on combined querysets ordered by unannotated columns.

### PR DESCRIPTION
**Pull request by iurisilvio**

### CODE REVIEW

File: tests/queries/test_qs_combinators.py [hunk 1]
```diff
@@ -234,6 +234,13 @@ def test_union_with_values_list_and_order_on_annotation(self):
             operator.itemgetter('num'),
         )
 
+    def test_union_multi_model_with_order(self):
+        ReservedName.objects.create(name='rn', order=0)
+        qs1 = Celebrity.objects.all()
+        qs2 = ReservedName.objects.all()
+        qs = qs1.union(qs2).values_list('name', flat=True)
+        self.assertEqual(qs.first(), 'rn')
```
Comment (felixxm) [inline tests/queries/test_qs_combinators.py:242]: `first()` is not crucial for this regression, so I've changed this to the `order_by()`.

File: django/db/models/sql/compiler.py [hunk 2]
```diff
@@ -320,18 +320,20 @@ def get_order_by(self):
...
                 continue
-            if col in self.query.annotations:
+
+            if self.query.combinator and self.select:
+                # Don't use the resolved annotation because other
+                # combinated queries might define it differently.
+                expr = F(col)
+                order_by.append((OrderBy(expr, descending=descending), False))
+                continue
+            elif col in self.query.annotations:
```
Comment (felixxm) [inline django/db/models/sql/compiler.py:330]: This will skip logic for `extra` columns. I added a test coverage and moved the fix to a different branch.

**Review by felixxm — COMMENTED** [review]
@iurisilvio Thanks :+1: I added release notes and pushed edits. Can you take a look?

**Comment by felixxm** [comment]
I've also noticed a separate issue with `extra()` and combined querysets (see ticket-32636).

File: tests/queries/test_qs_combinators.py (hunk 1 above)
Comment (iurisilvio) [reply tests/queries/test_qs_combinators.py]: Yes, the issue ia with order by.

**Comment by iurisilvio** [comment]
I think the release note has a typo: "when **if** was ordered" should be **"it"**.

**Comment by iurisilvio** [comment]
Your improvements looks good to me.

**Comment by felixxm** [comment]
> I think the release note has a typo: "when **if** was ordered" should be **"it"**.

Good catch :dart: 

Thanks for checking :+1:

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Pull request merged**